kernel/scheduler/src/lib.rs

```diff
@@ -53,15 +53,32 @@ pub fn init() -> Result<(), &'static str> {
         })
     }
 
-    #[cfg(not(target_arch = "x86_64"))] {
+    #[cfg(target_arch = "aarch64")] {
+        interrupts::enable_timer_interrupts(true, aarch64_timer_handler)
+    }
+
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))] {
         log::error!("TODO: scheduler::init() only supports registering a preemptive task switching timer interrupt on x86_64");
         Err("TODO: scheduler::init() only supports registering a preemptive task switching timer interrupt on x86_64")
     }
 }
 
+/// The handler for each CPU's local timer interrupt, used for preemptive task switching.
+#[cfg(target_arch = "aarch64")]
+extern "C" fn aarch64_timer_handler(_exc: &interrupts::ExceptionContext) -> interrupts::EoiBehaviour {
+    cpu_local_timer_tick_handler();
+
+    interrupts::EoiBehaviour::HandlerHasSignaledEoi
+}
+
 /// The handler for each CPU's local timer interrupt, used for preemptive task switching.
 #[cfg(target_arch = "x86_64")]
 extern "x86-interrupt" fn lapic_timer_handler(_stack_frame: x86_64::structures::idt::InterruptStackFrame) {
+    cpu_local_timer_tick_handler()
+}
+
+// Cross platform scheduling code
+fn cpu_local_timer_tick_handler() {
```

Comment:
this is nice, i like this design!

One request -- we should probably change the below "LAPIC_TIMER_TICKS` and the associated logging statement to something non-LAPIC-specific, just as you did with cpu_local_timer_handler

Reply:
> One request -- we should probably change the below "LAPIC_TIMER_TICKS` and the associated logging statement to something non-LAPIC-specific, just as you did with cpu_local_timer_handler

Np, done.

By the way, when I want to disable debugging statements like you did here, I usually just put an if statement rather than a config:
```rust
    if false { // stripped away on optimized builds
        use core::sync::atomic::{AtomicUsize, Ordering};
        static CPU_LOCAL_TIMER_TICKS: AtomicUsize = AtomicUsize::new(0);
        let _ticks = CPU_LOCAL_TIMER_TICKS.fetch_add(1, Ordering::Relaxed);
        log::info!("(CPU {}) CPU-LOCAL TIMER HANDLER! TICKS = {}", cpu::current_cpu(), _ticks);
    }
```
which has three benefits:
- it's also stripped away on release builds
- the content is still checked for errors
- last one is less important: to re-enable it, you just have to double-click false and type true, which is shorter than selecting the cfg stuff.

I didn't replace your way of doing it but I thought I'd mention it.

Reply:
oh that's a good way of doing it -- thanks for pointing that out! I wish rust supported syntax like `#[cfg(false)]`, but your point of it being checked during compilation is a key improvement.

You can definitely go ahead and change that here, i'll use that pattern elsewhere when i encounter the need for it.

Reply:
Another thing we should do (these can be done in a future PR) is to promote the `EoiBehavior` type to be used across architectures. I've frequently considered doing something like that on x86, so it'd be welcome there too, especially once we offer an arch-agnostic handler function type that is wrapped in an arch-specific entry point.